src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1354,14 +1354,17 @@ private void setItemToCache(final AuthenticationRequest request, AuthenticationR
             mTokenCacheStore.deleteIntersectingScope(key);
             
             TokenCacheItem cacheItem = new TokenCacheItem(request, result, result.getIsMultiResourceRefreshToken());
-            mTokenCacheStore.setItem(key, cacheItem);
-            
-            // Record idtoken with clientid as scope
-            if(request.isIdTokenRequest()){
-                cacheItem.setScope(new String[]{request.getClientId()});
-                key.setScope(new String[]{request.getClientId()});
-                mTokenCacheStore.setItem(key, cacheItem);
+
+            // If request contains open_id, convert it to client id.
+            final String[] scopesInRequest = request.getScope();
+            if (scopesInRequest != null && scopesInRequest.length == 1
```

Comment:
why would this massive if ever matter? In createAuthenticationResult you are putting correct values for scope already.